libraries/AP_HAL_ChibiOS/hwdef/common/halconf.h

```diff
@@ -478,6 +501,13 @@
 #define SPI_USE_WAIT                TRUE
 #endif
 
+/**
+ * @brief   Inserts an assertion on function errors before returning.
+ */
+#if !defined(SPI_USE_ASSERT_ON_ERROR) || defined(__DOXYGEN__)
+#define SPI_USE_ASSERT_ON_ERROR             TRUE
```

Comment:
I hope this will only raise asserts if we have enable asserts set.

Reply:
Yes, it only makes osalDbgAssert() be called which is guarded by asserts enablement